tests/e2e/test_spyre_cb.py

```diff
@@ -83,177 +93,555 @@ def test_cb_handling(
-        assert request_outputs[1].request_id == "2"  # req 2 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2  # requests 2 and 3 decoding now
-        assert request_outputs[0].request_id == "3"  # req 3 is decoding
-        assert request_outputs[1].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].finished  # request 2 is done
-        assert request_outputs[1].outputs[0].text == " 6 5 4 3 2 "
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 1
+    if not (steps_add_reqs == sorted(steps_add_reqs)
+            and steps_add_reqs[0] == 0):
+        raise ValueError(
+            "The list of steps where requests are added should be increasing "
+            "start with 0")
+
+    if not (checked_steps == sorted(checked_steps, key=lambda x: x["step"])
+            and len(checked_steps) == len(set(x["step"]
+                                              for x in checked_steps))):
+        raise ValueError(
+            "List of checked steps needs to be of increasing order of step")
+    # ------
+
+    # Setup the engine
+    engine_args = EngineArgs(model=model,
+                             tokenizer=model,
+                             max_model_len=2048,
+                             block_size=2048,
+                             max_num_seqs=max_num_seqs)
+    vllm_config = engine_args.create_engine_config()
+    executor_class = Executor.get_class(vllm_config)
+    engine_core = EngineCore(vllm_config=vllm_config,
+                             executor_class=executor_class,
+                             log_stats=False)
+    scheduler: ContinuousBatchingSpyreScheduler = engine_core.scheduler
+
+    # Create random requests of specified lengths and max_tokens
+    sorted_reqs_params = zip(steps_add_reqs, seqs_max_tokens, prompts_lengths)
+    requests: deque[tuple[int, EngineCoreRequest]] = deque()
+    for i, (add_step, max_tokens,
+            prompt_length) in enumerate(sorted_reqs_params):
+        # ignoring eos because we want to force the decoding to finish
+        # after max_tokens exactly
+        sampling_params = SamplingParams(max_tokens=max_tokens,
+                                         temperature=0.0,
+                                         ignore_eos=True)
+        request = create_random_request(request_id=i,
+                                        num_tokens=prompt_length,
+                                        sampling_params=sampling_params)
+        requests.append((add_step, request))
+
+    # if set, in-between steps are added as normal decode steps
+    if auto_check_basic_decode_steps:
+        checked_steps = augment_checked_steps(checked_steps)
+
+    # Run steps, until last step is reached (provided through parameter)
```

Comment:
`(provided through parameter)` -> is this a leftover comment from earlier implementation where the `last_step`  was given explicitly?

Reply:
ah yes, I should remove that